tests/test_basics.py

```diff
@@ -164,19 +164,6 @@ def test_func_with_recursive_data_within(data):
 
 
 class TestDecoration(unittest.TestCase):
-    def test_decorate_module(self):
```

Comment:
These two tests are testing an error on the user's side. They are trying to decorate  a non-class and non-callable object. Our typehint for `__call__` says we don't support it, so also IMO shouldn't build handling for it or test it.